### PR TITLE
Add ability to customise DevServices URL

### LINKED ISSUE #15579: Support in DevServices for additional JDBC URL parameters
**Description**
The DevServices feature currently doesn't yet support setting additional URL parameters for the created database. Some database (like Oracle and MySQL for example) allow specifying various parameters (e.g. time zone) as JDBC URL parameters. Currently the JDBC URL will be constructed by Testcontainers, so to specify URL parameters the Dev Service would have to call `org.testcontainers.containers.JdbcDatabaseContainer#withUrlParam()`.

Currently `DevServicesDatasourceProvider#startDatabase()` declares a parameter `Map<String, String> additionalProperties`. I am however not quite sure about the intended purpose of this parameter. While Testcontainers currently doesn't support specifying parameters (which would be passed to `java.sql.Driver#connect(String, Properties)`), this isn't really necessary either, because Dev Services doesn't delegate the Connection creation to Testcontainers, only the JDBC URL creation.

**Implementation ideas**
Possibly the `Map<String, String> additionalProperties` parameter could be (renamed and) repurposed. Or if I have misunderstood its purpose, possibly a new parameter could be introduced which would allow specifying additional JDBC URL parameters.

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #15579

### CODE REVIEW

File: extensions/devservices/h2/src/main/java/io/quarkus/devservices/postgresql/deployment/H2DevServicesProcessor.java [hunk 1]
```diff
@@ -27,11 +27,19 @@ public RunningDevServicesDatasource startDatabase(Optional<String> username, Opt
...
+                        additionalArgs.append(i.getKey());
+                        additionalArgs.append("=");
+                        additionalArgs.append(i.getValue());
+                    }
                     System.out
                             .println("[INFO] H2 database started in TCP server mode; server status: " + tcpServer.getStatus());
                     String connectionUrl = "jdbc:h2:tcp://localhost:" + tcpServer.getPort() + "/mem:"
                             + datasourceName.orElse("default")
-                            + ";DB_CLOSE_DELAY=-1";
+                            + ";DB_CLOSE_DELAY=-1" + additionalArgs.toString();
```
Comment (gsmet) [inline extensions/devservices/h2/src/main/java/io/quarkus/devservices/postgresql/deployment/H2DevServicesProcessor.java:42]: Ah ah, I was going to ask about `DB_CLOSE_DELAY`.

**Review by gsmet — APPROVED** [review]

File: extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java [hunk 2]
```diff
@@ -25,7 +25,24 @@ public RunningDevServicesDatasource startDatabase(Optional<String> username, Opt
...
+                    if (additionalArgs.length() > 0) {
+                        additionalArgs.append("&");
+                    }
+                    additionalArgs.append(i.getKey());
+                    additionalArgs.append("=");
+                    additionalArgs.append(i.getValue());
+                }
+                String jdbcUrl = container.getJdbcUrl();
+                if (additionalArgs.length() > 0) {
+                    if (jdbcUrl.contains("?")) {
```
Comment (knutwannheden) [inline extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java:39]: MSSQL is actually a bit special in this regard. The parameters are not separated using `?` and `&`. Please have a look at https://github.com/testcontainers/testcontainers-java/blob/master/modules/mssqlserver/src/main/java/org/testcontainers/containers/MSSQLServerContainer.java#L94-L95.

Comment (gsmet) [reply extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java:39]: @knutwannheden would you be willing to prepare a PR to adjust that?

Comment (knutwannheden) [reply extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java:39]: Yes, sure.

**Pull request merged**